python/pyspark/sql/pandas/group_ops.py

```diff
@@ -366,6 +366,10 @@ class PandasCogroupedOps:
     """
 
     def __init__(self, gd1: "GroupedData", gd2: "GroupedData"):
+        gel1 = gd1._jgd.groupingExprs().length()
```

Comment:
This may be a dumb question, but is it expensive to compute length()? if so I wonder if there's a cheaper way to detect this situation later

Reply:
Expensive because this is calling into the JVM and back, twice?

Later when `def applyInPandas` is called, Python calls into JVM anyway:

        jdf = self._gd1._jgd.flatMapCoGroupsInPandas(self._gd2._jgd, udf_column._jc.expr())

So this could be moved into `flatMapCoGroupsInPandas`.

Reply:
Oh OK this is just a call to a collection length(), not computing for example all of the unique keys and counting them... if so yeah fine

Reply:
The downside is that

    left.groupby("id", "k").cogroup(right.groupby("id"))

does not fail, only the subsequent `applyInPandas` does.